SmartDeviceLink/SDLIAPSession.m

```diff
@@ -64,9 +89,118 @@ - (BOOL)start {
 }
 
 - (void)stop {
-    [self stopStream:self.easession.outputStream];
-    [self stopStream:self.easession.inputStream];
+
+    if (!self.isDataSession){
+        [self stopStream:self.easession.outputStream];
+        [self stopStream:self.easession.inputStream];
+    }
+    else{
+        [self.ioStreamThread cancel];
+        
+        long lWait = dispatch_semaphore_wait(self.canceledSema, dispatch_time(DISPATCH_TIME_NOW, STREAM_THREAD_WAIT_SECS * NSEC_PER_SEC));
+        if (lWait == 0){
+            [SDLDebugTool logInfo:@"Stream thread canceled"];
+        }
+        else{
+           [SDLDebugTool logInfo:@"Error: failed to cancel stream thread"];
+        }
+        self.ioStreamThread = nil;
+        self.isDataSession = NO;
+    }
+    self.easession = nil;
+}
+
+- (void)sendData:(SessionSendHandler)sender{
+    @synchronized (self) {
+        self.sendBlock = sender;
+        self.isSendStarted = NO;
+    }
+    long lWait = dispatch_semaphore_wait(self.sendCompleteSema, dispatch_time(DISPATCH_TIME_NOW, STREAM_THREAD_WAIT_WRITE_SECS * NSEC_PER_SEC));
+    if (lWait != 0){
+        NSLog(@"ERROR! Failed to write data!!!");
+    }
+}
+
+- (void)accessoryEventLoop {
+    BOOL sendStarted;
+    SessionSendHandler sendBlock;
```

Comment:
These should be initialized with something, even if it's the default `NO` and `nil`.